code/modules/library/skill_learning/skill_station.dm

```diff
@@ -133,24 +141,26 @@
 	working = FALSE
 	work_timer = null
 	var/mob/living/carbon/carbon_occupant = occupant
-	var/obj/item/organ/brain/occupant_brain = carbon_occupant.getorganslot(ORGAN_SLOT_BRAIN)
-	if(QDELETED(carbon_occupant) || QDELETED(occupant_brain) || !(to_be_removed in occupant_brain.skillchips))
+
+	if(!to_be_removed.can_remove_safely())
+		to_chat(carbon_occupant,"<span class='notice'>Safety mechanisms activated! Skillchip cannot be safely removed.</span>")
 		return
-	to_be_removed.on_removal(carbon_occupant, silent=FALSE)
-	LAZYREMOVE(occupant_brain.skillchips, to_be_removed)
-	if(to_be_removed.removable)
-		carbon_occupant.put_in_hands(to_be_removed)
+
+	if(istype(carbon_occupant))
+		if(carbon_occupant.remove_skillchip(to_be_removed))
+			if(to_be_removed.removable)
+				carbon_occupant.put_in_hands(to_be_removed)
+			else
+				qdel(to_be_removed)
+
+			to_chat(carbon_occupant, "<span class='notice'>Operation complete!</span>")
+		else
+			to_chat(carbon_occupant,"<span class='notice'>Failed to remove skillchip!</span>")
```

Comment:
Repeat `update_icon` and `SStgui.update_uis(src)` if needed, but turn these into early returns, please.